mathesar_ui/src/components/sheet/types.ts

```diff
@@ -0,0 +1,9 @@
+import type { Column } from '@mathesar/stores/table-data/types';
+import type { AbstractType } from '@mathesar/stores/abstract-types/types';
+import type { getCellCap } from '@mathesar/components/cell/utils';
+
+export interface SheetColumn {
+  column: Column;
+  abstractTypeOfColumn: AbstractType;
+  cellCap: ReturnType<typeof getCellCap>; // CAP refers to ComponentAndProps
```

Comment:
As mentioned in my other comment, I don't like this abbreviation. I'd rather write `ComponentAndProps` for better readability.

If there's an important reason for keeping the "CAP" naming convention though, the code comment would be better as a JSDoc comment above the `cellCap` property so that it will be semantically associated and appear in editor tooltips elsewhere.